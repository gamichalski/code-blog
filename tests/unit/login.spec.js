import { shallowMount } from "@vue/test-utils";
import Login from "@/views/login/Index";

const factory = (data) =>
  shallowMount(Login, {
    data: () => ({ ...data }),
  });

describe("Login", () => {
  it("Deve ser um instância do vue", () => {
    const wrapper = factory();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it("Deve definir o primeiro elemento input com o valor de email do usuário passado", () => {
    const user = { username: "teste@codenation.com" };
    const wrapper = factory({ user });

    const inputs = wrapper.findAll("input");
    const username = inputs.at(0);

    expect(username.element.value).toEqual("teste@codenation.com");
  });

  it("Deve definir o segundo elemento input com o valor de senha do usuário passado", () => {
    const user = { password: "123456" };
    const wrapper = factory({ user });

    const inputs = wrapper.findAll("input");
    const username = inputs.at(1);

    expect(username.element.value).toEqual("123456");
  });

  it("Deve ser chamado a função de login quando o click no botão de 'Entrar' for disparado", () => {
    const wrapper = factory();
    wrapper.vm.login = jest.fn().mockImplementation(() => Promise.resolve());

    wrapper.find("button").trigger("click");

    expect(wrapper.vm.login).toBeCalled();
  });

  it("Deve ser redirecionado para a tela de posts quando a função de login retornar sucesso", async () => {
    const wrapper = factory();

    wrapper.vm.login = jest.fn().mockImplementation(() => Promise.resolve());
    wrapper.vm.$router = { push: jest.fn() };

    await wrapper.vm.onClickLogin();
    const [args] = wrapper.vm.$router.push.mock.calls[0];

    expect(args.name).toEqual("Posts");
  });

  it("Deve ser limpa a senha ao retornar erro na função de login", async () => {
    const wrapper = factory();

    wrapper.vm.login = jest.fn().mockImplementation(() => Promise.reject());

    await wrapper.vm.onClickLogin();

    expect(wrapper.vm.user.password).toEqual("");
  });

  it("Deve ter um snapshot igual", () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
