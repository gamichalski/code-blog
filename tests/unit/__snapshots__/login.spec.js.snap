// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Deve ter um snapshot igual 1`] = `
<section class="hero is-fullheight is-dark">
  <div class="hero-body">
    <div class="columns is-fullwidth">
      <div class="column is-4 is-offset-4">
        <div class="box">
          <h1 class="title has-text-centered has-text-dark">Login</h1>
          <div class="field"><label class="label">Email</label>
            <div class="control"><input type="text" class="input"></div>
          </div>
          <div class="field"><label class="label">Senha</label>
            <div class="control"><input type="password" class="input"></div>
          </div>
          <div class="control"><button class="button is-dark is-fullwidth">Entrar</button></div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
